### ISSUE TITLE
Unable to play any files

### ISSUE THREAD

**Issue by fossxplorer** [owncloud/music#432]
I have had quite some issues with Music master  on oC stable8.1. It doesn't work for some users of the instance and for the one where it works i get the following in the console:
TypeError: OC.Backbone is undefined

I'm unable to play any music files via web UI.

Another issue is that i get  500 for all requets to /index.php/apps/music/api/collection which gives blank page with spinner spinning for ever.

occ music:scan user1 -vvv just gives:
Start scan for  user1

Webserver log:
"server: , request: "GET /apps3/music/js/vendor/underscore/underscore-min.map HTTP/1.1 404"

@MorrisJobke

**Comment by MorrisJobke** [comment]
> Another issue is that i get 500 for all requets to /index.php/apps/music/api/collection which gives blank page with spinner spinning for ever.

Is there anything in the owncloud.log in your data directory?

**Comment by fossxplorer** [comment]
app":"music","message":"JS: Processed more files than available 384\/4","level":0,"time":"2015-09-28T08:53:44+00:00","method":"POST","url":"\/index.php\/apps\/music\/api\/log

{"reqId":"0xoN\/rIdsPwwoZkVtoVo","remoteAddr":"","app":"index","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\NotFoundException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\//oc/apps3/music/controller/apicontroller.php(116): OCA\Music\Db\Track->toCollection(Object(OC\URLGenerator), Object(OC\Files\Node\Folder))\\n#1 [internal function]: OCA\Music\Controller\ApiController->collection()\\n#2 /var/www//oc//lib/private/appframework/http/dispatcher.php(159): call_user_func_array(Array, Array)\\n#3 /var/www//oc//lib/private/appframework/http/dispatcher.php(89): OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\Music\Controller\ApiController), 'collection')\\n#4 /var/www//oc//lib/private/appframework/app.php(108): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Music\Controller\ApiController), 'collection')\\n#5 /var/www//oc//lib/private/appframework/routing/routeactionhandler.php(45): OC\AppFramework\App::main('ApiController', 'collection', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)\\n#6 [internal function]: OC\AppFramework\routing\RouteActionHandler->__invoke(Array)\\n#7 /var/www//oc//lib/private/route/router.php(274): call_user_func(Object(OC\AppFramework\routing\RouteActionHandler), Array)\\n#8 /var/www//oc//lib/base.php(873): OC\Route\Router->match('/apps/music/api...')\\n#9 /var/www//oc//index.php(40): OC::handleRequest()\\n#10 {main}\",\"File\":\"/var/www//oc/apps3/music/db/track.php\",\"Line\":97}","level":4,"time":"2015-09-28T08:53:43+00:00","method":"GET","url":"\/index.php\/apps\/music\/api\/collection"}

**Comment by select** [comment]
I have a similar/the same problem, I think it happened when I attached an external drive with sftp and then lost the connection to it. Removing the drive from the admin did not help. 

Other users that did not have this directory mounted are not affected by this problem

Exception: {"Exception":"OCP\\Files\\NotFoundException","Message":"","Code":0,"Trace":"#0 \/var\/www\/owncloud\/apps\/music\/controller\/apicontroller.php(116): OCA\\Music\\Db\\Track->toCollection(Object(OC\\URLGenerator), Object(OC\\Files\\Node\\Folder))\n#1 [internal function]: OCA\\Music\\Controller\\ApiController->collection()\n#2 \/var\/www\/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(159): call_user_func_array(Array, Array)\n#3 \/var\/www\/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Music\\Controller\\ApiController), 'collection')\n#4 \/var\/www\/owncloud\/lib\/private\/appframework\/app.php(108): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Music\\Controller\\ApiController), 'collection')\n#5 \/var\/www\/owncloud\/lib\/private\/appframework\/routing\/routeactionhandler.php(45): OC\\AppFramework\\App::main('ApiController', 'collection', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n#6 [internal function]: OC\\AppFramework\\routing\\RouteActionHandler->__invoke(Array)\n#7 \/var\/www\/owncloud\/lib\/private\/route\/router.php(274): call_user_func(Object(OC\\AppFramework\\routing\\RouteActionHandler), Array)\n#8 \/var\/www\/owncloud\/lib\/base.php(873): OC\\Route\\Router->match('\/apps\/music\/api...')\n#9 \/var\/www\/owncloud\/index.php(40): OC::handleRequest()\n#10 {main}","File":"\/var\/www\/owncloud\/apps\/music\/db\/track.php","Line":97}

**Comment by select** [comment]
I thought resetting the db might help, but i cannot get it to work

```
# sudo -u www-data php occ music:reset-database
Clean up relations (album/artist and playlist/track)
PHP Fatal error:  Call to undefined method OC\DB\Connection::prepareQuery() in /var/www/oc/apps/music/command/resetdatabase.php on line 85
```

**Comment by mattfletcher** [comment]
I'm getting a very similar message.

    # sudo -u admin php occ music:reset-database --all
    Drop tables for all users
    Clean up relations (album/artist and playlist/track)
    PHP Fatal error:  Call to undefined method OC\DB\Connection::prepareQuery() in /home/admin/web/cloud.torrix.uk/public_html/apps/music/command/resetdatabase.php on line 85

My music shows up in the Music app, and when I click on a track, the play icon appears next to the track, but there is no sound.

**Comment by MorrisJobke** [comment]
You are all on ownCloud 8.1?

**Comment by mattfletcher** [comment]
ownCloud 8.1.3 (stable) - freshly installed yesterday using the setup-owncloud.php script.

**Comment by select** [comment]
me too $OC_VersionString = '8.1.3'; manually updated from the previous version

**Comment by MorrisJobke** [comment]
@select @mattfletcher Your issues are fixed with #411

**Comment by MorrisJobke** [comment]
@select @fossxplorer The problem with the FileNotFound exception is, that there are invalid entries in the database. I need to check how to improve this handling then.

**Comment by BigButt** [comment]
Same problem here. ownCloud 8.1.3 and music 0.3.8.
Can't play a file.

**Comment by derula** [comment]
Getting the same issue with ownCloud 8.1.4, Music 0.3.8.

The console shows "TypeError: OC.Backbone is undefined" on page-load.

Tested in Firefox and Chromium.

Clicking a song results in the playing indicator showing next to the file name, but there is no audio.

Also, no further message pops up in the console upon clicking a file.

**Comment by MorrisJobke** [comment]
Gotcha...https://github.com/owncloud/music/blob/e5f1d52a5aac617974001b5723afb762044d344f/js/app/playerwrapper.js#L11

Let me fix this.

**Comment by MorrisJobke** [comment]
Fix for this is in #446
